kernel/src/hil/adc.rs

```diff
@@ -1,14 +1,20 @@
-//! Simplest attempt at an ADC interface.
-//!
-//!  Author: Philip Levis <pal@cs.stanford.edu>
-//!  Date: June 10, 2015
-//!
-
-pub trait Request {
-    fn sample_done(&self, val: u16, request: &'static Request);
+pub trait Client {
+    fn sample_done(&self, sample: u16);
 }
 
-pub trait AdcInternal {
-    fn initialize(&'static mut self) -> bool;
-    fn sample(&self, channel: u8, callback: &'static Request) -> bool;
+pub trait AdcSingle {
+    fn initialize(&self) -> bool;
+    fn sample(&self, channel: u8) -> bool;
+    // fn cancel_sample(&self) -> bool;
 }
+
+// pub trait Frequency {
```

Comment:
I think the comment out lines are functions that we decided should be part of the HIL but haven't been fulfilled yet. If we delete them, I'd give it like 90% odds we forget they still need to be done.